src/compiler/types.ts

```diff
@@ -4967,8 +4967,8 @@ export interface TypeChecker {
     typeToTypeNode(type: Type, enclosingDeclaration: Node | undefined, flags: NodeBuilderFlags | undefined): TypeNode | undefined;
     /** @internal */ typeToTypeNode(type: Type, enclosingDeclaration: Node | undefined, flags: NodeBuilderFlags | undefined, tracker?: SymbolTracker): TypeNode | undefined; // eslint-disable-line @typescript-eslint/unified-signatures
     /** Note that the resulting nodes cannot be checked. */
-    signatureToSignatureDeclaration(signature: Signature, kind: SyntaxKind, enclosingDeclaration: Node | undefined, flags: NodeBuilderFlags | undefined): SignatureDeclaration & {typeArguments?: NodeArray<TypeNode>} | undefined;
-    /** @internal */ signatureToSignatureDeclaration(signature: Signature, kind: SyntaxKind, enclosingDeclaration: Node | undefined, flags: NodeBuilderFlags | undefined, tracker?: SymbolTracker): SignatureDeclaration & {typeArguments?: NodeArray<TypeNode>} | undefined; // eslint-disable-line @typescript-eslint/unified-signatures
+    signatureToSignatureDeclaration(signature: Signature, kind: SignatureDeclaration["kind"], enclosingDeclaration: Node | undefined, flags: NodeBuilderFlags | undefined): SignatureDeclaration & {typeArguments?: NodeArray<TypeNode>} | undefined;

```

Comment:
Uh oh: [Playground Link](https://www.typescriptlang.org/play?ts=4.9.5#code/KYOwrgtgBAygniALgQwB4GkCWIAmUDeAsAFBRlQBCANgPYDGA1gDQnlQBiYIdimNIAEWB0qyAE7Je-FsQC+JEtkTAxAM2R1gUAMIALYQxUFW5AM6YA5iElgxwACo0Yl64lvAhI8ZL4gAFAzYOABcsAgoGEEAlKEAbjSYOCZk5lY2do7OaW52nqISUiAATAFBofBIaFi4MVDxiSTyxCQ4wvlaqlw8vlCpru6ZLukebd6Fpbjl4VVBAHSc3IV5Y7619UnNxHT8pohQdPqMKqF6BkYAvMak5AD0N1AA6rpwUDg0wKZQiLqYn9sQAAdMFRgAB+XpDHIOJyQ9zLAo9YCoAFUTB0TCIKgvUzIOCfDFQfhYr7IQymZJQO4cLpLUYI-izCl9YaDbJwuk+aRM2EZGFs3IcwpFUITHBRKDnAB8EP50Ky-QFXnp-kCNRksiAA)

Reply:
Seems to be fixed in 5.0.

Reply:
Yep, it's method bivariance.  [This one errors as expected.](https://www.typescriptlang.org/play?ts=4.9.5#code/KYOwrgtgBAygniALgQwB4GkCWIAmUDeAsAFBRlQBCANgPYDGA1gDQnlQBiYIdimNIAEWB0qyAE7Je-FsQC+JEtkTAxAM2R1gUAMIALYQxUFW5AM6YA5iElgxwACo0Yl64lvAhI8ZL4gAXFAAFAzYOAHwSGhYuACUUAC8AHxQAG40mDgmZOZWNnaOzrludp6iElIgAEwBwaHhCCgYoXFJqemZcgrEOMJlWqpcPL5QOa7uBS55Hr3eFbW49ZFNuAB0nNwVpbO+MQFpGV10-KaIUHT6jCoBegZG8cak5AD0T1AA6rpwUDg0wKZQiF0mH+RwgAAdMFRgAB+EaTYoOJzw9xbcrDYCoMFUTB0TCIKhfUzIOD-PFQfgEgHIQymLJQF4cQabGZo-grOmjKYTIoolk+aQc5H5JE8kp8irVIIhWIJZKchHcsZiryskDzHAxGSyIA)